script.js

```diff
@@ -30,3 +30,120 @@ navitems.forEach((item) => {
     close.classList.remove('hide');
   });
 });
+
+// FORM VALIDATION;
+
+const userNameE = document.querySelector('#myName');
+const form = document.querySelector('#myForm');
+const emailAdress = document.querySelector('#myEmail');
+
+const isRequired = (value) => (value !== '');
+const isBetween = (length, min, max) => (!(length < min || length > max));
+
+const isEmailValid = (email) => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+};
+
+const showError = (input, message) => {
+  // get the form-field element
+  const formField = input.parentElement;
+  // add the error class
+  formField.classList.remove('success');
+  formField.classList.add('error');
+
+  // show the error message
+  const error = formField.querySelector('small');
+  error.textContent = message;
+};
+
+const showSuccess = (input) => {
+  // get the form-field element
+  const formField = input.parentElement;
+
+  // remove the error class
+  formField.classList.remove('error');
+  formField.classList.add('success');
+
+  // hide the error message
+  const error = formField.querySelector('small');
+  error.textContent = '';
+};
+
+const checkUsername = () => {
+  let valid = false;
+
+  const min = 3;
+  const max = 25;
+
+  const username = userNameE.value.trim();
+
+  if (!isRequired(username)) {
+    showError(userNameE, 'Username cannot be blank.');
+  } else if (!isBetween(username.length, min, max)) {
+    showError(userNameE, `Username must be between ${min} and ${max} characters.`);
+  } else {
+    showSuccess(userNameE);
+    valid = true;
+  }
+  return valid;
+};
+
+const checkEmail = () => {
+  let valid = false;
+  const email = emailAdress.value.trim();
+  if (!isRequired(email)) {
+    showError(emailAdress, 'Email cannot be blank.');
+  } else if (!isEmailValid(email)) {
+    showError(emailAdress, 'Email is not valid.');
+  } else {
+    showSuccess(emailAdress);
+    valid = true;
+  }
+  return valid;
+};
```

Comment:
You did well with your contact form validation, however, according to the project requirement you are to implement the following :
The contact form contains at least one validation checking if the email is written in all lowercase:

1. If a user submits the contact form with the email field containing uppercase characters an error message should show up.

2. If a user submits the contact form with the email field containing only lowercase characters the form is sent normally.

Although an error shows when a user inputs an uppercase letter, the normal behavior of the form is that it shouldn't submit it except all characters are in lowercase, please write validation for this as there is none at the moment which makes your form to submit 😄